Formula/kittycad.rb

```diff
@@ -0,0 +1,30 @@
+class Kittycad < Formula
+  desc "Command-line tool for interacting with KittyCAD"
+  homepage "https://github.com/kittycad/cli"
+  url "https://github.com/KittyCAD/cli/archive/refs/tags/v0.0.1.tar.gz"
+  sha256 "16fdeaf14d74ac76ed9dca955c167ebeef8557ded0f1fa3eb712f2c49f1b4dd9"
+  license "MIT"
+
+  head "https://github.com/kittycad/cli.git", branch: "main"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "go" => :build
+
+  def install
+    system "make", "build", "gen-man"
+
+    bin.install "cli" => "kittycad"
+    man1.install Dir["generated_docs/man/kittycad*.1"]
+    (bash_completion/"kittycad").write `#{bin}/kittycad completion -s bash`
+    (fish_completion/"kittycad.fish").write `#{bin}/kittycad completion -s fish`
+    (zsh_completion/"_kittycad").write `#{bin}/kittycad completion -s zsh`
+  end
+
+  test do
+    assert_match "kittycad version #{version}", shell_output("#{bin}/kittycad --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
ah will do thanks!